bindings/matrix-sdk-ffi/src/room/mod.rs

```diff
@@ -1139,6 +1139,59 @@ impl Room {
 
         Ok(Arc::new(RoomPreview::new(AsyncRuntimeDropped::new(client), room_preview)))
     }
+
+    /// Toggle a MSC4306 subscription to a thread in this room, based on the
+    /// thread root event id.
+    ///
+    /// If `subscribed` is `true`, it will subscribe to the thread, with a
+    /// precision that the subscription was manually requested by the user
+    /// (i.e. not automatic).
+    ///
+    /// If the thread was already subscribed to (resp. unsubscribed from), while
+    /// trying to subscribe to it (resp. unsubscribe from it), it will do
+    /// nothing, i.e. subscribing (resp. unsubscribing) to a thread is an
+    /// idempotent operation.
+    pub async fn toggle_thread_subscription(
```

Comment:
"toggle" to me means flip to true if it's false or to false if it's true.

So I'd suggest `set_thread_subscription`.